File: 888.fair-candy-swap.0.kt
/*
 * @lc app=leetcode id=888 lang=kotlin
 *
 * [888] Fair Candy Swap
 */
class Solution_fairCandySwap_0 {
    fun fairCandySwap(A: IntArray, B: IntArray): IntArray {
        val sumA = A.sum()
        val sumB = B.sum()
        A.forEach { vA ->
            val vB = B.find { vB -> (vA - vB) * 2 == sumA - sumB }
            if (vB != null) {
                return intArrayOf(vA, vB)
            }
        }
        return intArrayOf()
    }
}

File: 888.fair-candy-swap.0.go
/*
 * @lc app=leetcode id=888 lang=golang
 *
 * [888] Fair Candy Swap
 */
func fairCandySwap(A []int, B []int) []int {
    sumA := 0
    sumB := 0
    for _, n := range A {
        sumA += n
    }
    for _, n := range B {
        sumB += n
    }
    sum := sumA + sumB
    exchange := sum / 2 - sumA
    for _, n1 := range A {
        for _, n2 := range B {
            if n1 + exchange == n2 {
                return []int{n1, n2}
            }
        }
    }

    return []int{}
}
